evm/its.js

```diff
@@ -0,0 +1,723 @@
+                throw new Error(`Invalid sourceAddress: ${sourceAddress}`);
+            }
+
+            if (!isValidCalldata(payload)) {
+                throw new Error(`Invalid payload: ${payload}`);
+            }
+
+            const tx = await interchainTokenService.expressExecute(commandID, sourceChain, sourceAddress, payload);
+            printInfo('expressExecute tx', tx.hash);
+
+            const receipt = await tx.wait(chain.confirmations);
+
+            const eventEmitted = wasEventEmitted(receipt, interchainTokenService, 'ExpressExecuted');
+
+            if (!eventEmitted) {
+                printWarn('Event not emitted in receipt.');
+            }
+
+            break;
+        }
+
+        case 'interchainTransfer': {
+            const isPaused = await interchainTokenService.paused();
+
+            if (isPaused) {
+                throw new Error(`${action} invalid while service is paused.`);
+            }
+
+            const { tokenId, destinationChain, destinationAddress, amount, metadata } = options;
+
+            if (!isValidTokenId(tokenId)) {
+                throw new Error(`Invalid tokenId value: ${tokenId}`);
+            }
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            if (!isString(destinationChain)) {
+                throw new Error(`Invalid destinationChain: ${destinationChain}`);
+            }
+
+            if (!isString(destinationAddress)) {
+                throw new Error(`Invalid destinationAddress: ${destinationAddress}`);
+            }
+
+            if (!isValidNumber(amount)) {
+                throw new Error(`Invalid token amount: ${amount}`);
+            }
+
+            if (!isValidCalldata(metadata)) {
+                throw new Error(`Invalid metadata: ${metadata}`);
+            }
+
+            const tx = await interchainTokenService.interchainTransfer(
+                tokenIdBytes32,
+                destinationChain,
+                destinationAddress,
+                amount,
+                metadata,
+            );
+            printInfo('interchainTransfer tx', tx.hash);
+
+            const receipt = await tx.wait(chain.confirmations);
+
+            const eventEmitted =
+                wasEventEmitted(receipt, interchainTokenService, 'InterchainTransfer') ||
+                wasEventEmitted(receipt, interchainTokenService, 'InterchainTransferWithData');
+
+            if (!eventEmitted) {
+                printWarn('Event not emitted in receipt.');
+            }
+
+            break;
+        }
+
+        case 'callContractWithInterchainToken': {
+            const isPaused = await interchainTokenService.paused();
+
+            if (isPaused) {
+                throw new Error(`${action} invalid while service is paused.`);
+            }
+
+            const { tokenId, destinationChain, destinationAddress, amount, data } = options;
+
+            if (!isValidTokenId(tokenId)) {
```

Comment:
Lot of code duplication. Some checks you can perform in the beginning if the option was provided, or create reusable functions